src/argilla/server/schemas/v1/records.py

```diff
@@ -13,41 +13,153 @@
+            else:
+                return default
 
+        return super().get(key, default)
 
-class Response(BaseModel):
+
+class Record(BaseModel):
     id: UUID
-    values: Optional[Dict[str, ResponseValue]]
-    status: ResponseStatus
-    user_id: UUID
+    fields: Dict[str, Any]
+    metadata: Optional[Dict[str, Any]]
+    external_id: Optional[str]
+    # TODO: move `responses` to `response` since contextualized endpoint will contains only the user response
+    # response: Optional[Response]
+    responses: Optional[List[Response]]
+    suggestions: Optional[List[Suggestion]]
+    vectors: Optional[Dict[str, List[float]]]
+    dataset_id: UUID
     inserted_at: datetime
     updated_at: datetime
 
     class Config:
         orm_mode = True
+        getter_dict = RecordGetterDict
+
+
+class RecordCreate(BaseModel):
+    fields: Dict[str, Any]
+    metadata: Optional[Dict[str, Any]]
+    external_id: Optional[str]
+    responses: Optional[List[UserResponseCreate]]
+    suggestions: Optional[List[SuggestionCreate]]
+    vectors: Optional[Dict[str, List[float]]]
+
+    @validator("responses")
+    def check_user_id_is_unique(cls, values: Optional[List[UserResponseCreate]]) -> Optional[List[UserResponseCreate]]:
+        if values is None:
+            return values
 
+        user_ids = []
+        for value in values:
+            if value.user_id in user_ids:
+                raise ValueError(f"'responses' contains several responses for the same user_id={str(value.user_id)!r}")
+            user_ids.append(value.user_id)
 
-class ResponseCreate(BaseModel):
-    values: Optional[Dict[str, ResponseValueCreate]]
-    status: ResponseStatus
+        return values
 
 
 class RecordUpdate(UpdateSchema):
     metadata_: Optional[Dict[str, Any]] = Field(None, alias="metadata")
     suggestions: Optional[List[SuggestionCreate]] = None
     vectors: Optional[Dict[str, List[float]]]
+
+
+class RecordUpdateWithId(RecordUpdate):
+    id: UUID
+
+
+class RecordIncludeParam(BaseModel):
+    relationships: Optional[List[RecordInclude]] = Field(None, alias="keys")
+    vectors: Optional[List[str]] = Field(None, alias="vectors")
+
+    @root_validator(skip_on_failure=True)
+    def check(cls, values: Dict[str, Any]) -> Dict[str, Any]:
+        relationships = values.get("relationships")
+        if not relationships:
+            return values
+
+        vectors = values.get("vectors")
+        if vectors is not None and len(vectors) > 0 and RecordInclude.vectors in relationships:
+            # TODO: once we have a exception handler for ValueError in v1, remove HTTPException
+            # raise ValueError("Cannot include both 'vectors' and 'relationships' in the same request")
+            raise HTTPException(
+                status_code=422,
+                detail="'include' query param cannot have both 'vectors' and 'vectors:vector_settings_name_1,vectors_settings_name_2,...'",
+            )
```

Comment:
We can raise a `ValueError` and fastapi will automatically handle as a 422 status response
```suggestion
            raise ValueError("'include' query param cannot have both 'vectors' and 'vectors:vector_settings_name_1,vectors_settings_name_2,...'",
            )
```